monkestation/code/modules/oldstation_idconsole/oldstation_idconsole.dm

```diff
@@ -0,0 +1,154 @@
+//i'd rather type suicide in the command bar than put everything in separate files
+
+//charlie station uses it's own machine for this since all it does is turn the id into an old id
+/obj/machinery/oldpdapainter
+	name = "Charlie Station ID Painter"
+	desc = "A painting machine that can be used to paint IDs. To use, simply insert the ID. It looks old and dusty.."
+	icon = 'icons/obj/pda.dmi'
+	icon_state = "pdapainter"
+	density = TRUE
+	max_integrity = 200
+	use_power = IDLE_POWER_USE
+	idle_power_usage = BASE_MACHINE_IDLE_CONSUMPTION * 0.02
+	active_power_usage = BASE_MACHINE_ACTIVE_CONSUMPTION * 0.05
+	//dont turn these ids into charlie station ids please, this might be redundant
+	var/list/blacklisted_types = list()
+
+/obj/machinery/oldpdapainter/Initialize()
+	blacklisted_types += typesof(/obj/item/card/id/away)
+	blacklisted_types += typesof(/obj/item/card/id/advanced/silver)
+	blacklisted_types += typesof(/obj/item/card/id/advanced/gold)
+	blacklisted_types += typesof(/obj/item/card/id/advanced/chameleon)
+	. = ..()
```

Comment:
```suggestion
	//dont turn these ids into charlie station ids please, this might be redundant
	var/static/list/blacklisted_typecache = typecacheof(list(
		/obj/item/card/id/away,
		/obj/item/card/id/advanced/silver,
		/obj/item/card/id/advanced/gold,
		/obj/item/card/id/advanced/chameleon,
	))
```
use a typecache, like so